File: Controls/EditorA/filepicker/src/main/java/com/kelfan/filepicker/FilterComposite.java
package com.kelfan.filepicker;

import java.io.File;
import java.io.FileFilter;
import java.io.Serializable;
import java.util.ArrayList;

/**
 * Created by Administrator on 7/03/2018.
 */

public class FilterComposite implements FileFilter, Serializable {

    private ArrayList<FileFilter> mFilters;

    public FilterComposite(ArrayList<FileFilter> filters) {
        mFilters = filters;
    }

    @Override
    public boolean accept(File f) {
        for (FileFilter filter : mFilters) {
            if (!filter.accept(f)) {
                return false;
            }
        }

        return true;
    }
}

File: MyProject/07KotlinEditor/app/src/main/java/com/kelfan/kotlineditor/ui/filepicker/FilterComposite.kt
package com.kelfan.kotlineditor.ui.filepicker

import java.io.File
import java.io.FileFilter
import java.io.Serializable
import java.util.ArrayList

/**
 * Created by Administrator on 2/02/2018.
 */

class FilterComposite(private val fileFilters: ArrayList<FileFilter>) : FileFilter, Serializable {

    override fun accept(file: File): Boolean {
        for (filter in fileFilters) {
            if (!filter.accept(file)) {
                return false
            }
        }

        return true
    }
}
